server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceAggregatorFactory.java

```diff
@@ -43,13 +43,17 @@ public ValuesSourceAggregatorFactory(String name, ValuesSourceConfig<VS> config,
     @Override
     public Aggregator createInternal(Aggregator parent, boolean collectsFromSingleBucket,
             List<PipelineAggregator> pipelineAggregators, Map<String, Object> metaData) throws IOException {
-        VS vs = config.toValuesSource(context.getQueryShardContext());
+        VS vs = config.toValuesSource(context.getQueryShardContext(), this::resolveMissingAny);
         if (vs == null) {
             return createUnmapped(parent, pipelineAggregators, metaData);
         }
         return doCreateInternal(vs, parent, collectsFromSingleBucket, pipelineAggregators, metaData);
     }
 
+    protected ValuesSource resolveMissingAny(Object missing) {
```

Comment:
Let's add some javadocs on this method so it's clear what this is for (and why a subclass might want to override it)

Reply:
Done.

Reply:
:tada:  awesome thanks